daemon/cluster/cluster.go

```diff
@@ -186,7 +190,10 @@ func New(config Config) (*Cluster, error) {
 		if errors.Cause(c.err) == ErrSwarmLocked {
 			return c, nil
 		}
-
+		if errors.Cause(c.err).Error() == (x509.CertificateInvalidError{Reason: x509.Expired}).Error() {
```

Comment:
Why not convert to `CertificateInvalidError` and check that `cErr.Reason == x509.Expired`?
Seems cleaner than the string comparison.